nifi-extension-bundles/nifi-kafka-bundle/nifi-kafka-processors/src/main/java/org/apache/nifi/kafka/processors/PublishKafka.java

```diff
@@ -366,13 +376,51 @@ public List<ConfigVerificationResult> verify(final ProcessContext context, final
         return verificationResults;
     }
 
+
+    @OnStopped
+    public void onStopped() {
+        // Ensure that we close all Producer services when stopped
+        KafkaProducerService service;
+
+        while ((service = producerServices.poll()) != null) {
+            service.close();
+        }
+    }
+
     @Override
     public void onTrigger(final ProcessContext context, final ProcessSession session) throws ProcessException {
         final List<FlowFile> flowFiles = PublishKafkaUtil.pollFlowFiles(session);
         if (flowFiles.isEmpty()) {
             return;
         }
 
+        final KafkaProducerService producerService = getProducerService(context);
```

Comment:
This call can throw a Kafka `TimeoutException` causing uncaught exception handling and yielding, which does not seem like the appropriate behavior given the property to configure failing handling.

Reply:
On further consideration, this aligns with the current behavior where failure to get a Kafka Producer also results in an exception and yielding.